### PR TITLE
Support source mapping of stack traces in the Debug Console #6

### PR DESCRIPTION
**Pull request by nojvek**
Will add some tests tomorrow. Still figuring out how tests work with sourcemap mocks.

Let me know if the code section looks good by you.

@auchenberg @roblourens

### CODE REVIEW

**Comment by roblourens** [comment]
Thanks, this looks great! Tests are always welcome.

File: src/chrome/chromeDebugAdapter.ts [hunk 1]
```diff
@@ -740,10 +740,37 @@ export abstract class ChromeDebugAdapter implements IDebugAdapter {
...
+        const exceptionLines = formattedException.split(/\r?\n/);
+
+        for (let i = 0, len = exceptionLines.length; i < len; ++i) {
+            const line = exceptionLines[i];
+            const matches = line.match(/^\s+at (.*?)\s*\(?([^ ]+\.js):(\d+):(\d+)\)?$/);
+
+            if (matches) {
+                const fnName = matches[1];
+                const path = matches[2];
+                const lineNum = parseInt(matches[3]);
```
Comment (auchenberg) [inline src/chrome/chromeDebugAdapter.ts]: TSLint failes due to missing radex param.

**Comment by nojvek** [comment]
Just installed `tslint` plugin for vscode. Just realized that that it now has `tslint.autofix` like eslint. 

I press save and everything fixes itself. So awesome 😎 🎉

**Comment by nojvek** [comment]
@roblourens I am trying to get the unit test for exceptions to work.

Firing `mockEventEmitter.emit('Runtime.exceptionThrown', ...` but that never ends up calling `onExceptionThrown` even thought the handler is run `     this.chrome.Runtime.onExceptionThrown(params => this.onExceptionThrown(params));
`. Weird, am I doing something obviously wrong that I can't see?

My test strategy is to fire a bunch of `Runtime.exceptionThrown` events, mock the sourcemapper and pathmapper and check the final exception string sent to client is source mapped.

**Comment by roblourens** [comment]
Looks like there's a typo in my mock event handling code - https://github.com/Microsoft/vscode-chrome-debug-core/blob/master/test/mocks/debugProtocolMocks.ts#L51 - should be `Runtime.exceptionThrown` not `onExceptionThrown`.

**Comment by nojvek** [comment]
@roblourens Thanks for the pointer. 

Making mocks work properly was a mission but I think I have it all working now.

**Comment by nojvek** [comment]
Tested on dummy and webpack projects. `I'm lovin it` 

image

**Comment by roblourens** [comment]
Awesome, thanks for the tests

**Pull request merged**